### PR TITLE
typecheck adapters

### LINKED ISSUE #625: Type checking in adapters
# Issue

The adapters don't have TypeScript. This leads two two big issues.

1. we introduce bugs which would be caught with type checking https://github.com/sveltejs/kit/pull/600/files#r600142752
2. VS Code is painful to use because everything is red squigglies

Screenshot from 2021-03-23 20-41-29

# Possible solutions

https://github.com/microsoft/TypeScript/issues/33079#issuecomment-738033267

### PR DESCRIPTION
**Pull request by Rich-Harris**
Closes #625

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

File: packages/adapter-cloudflare-workers/index.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ export default function () {
 	};
 }
 
-/** @param {import('@sveltejs/kit').Builder} builder */
+/** @param {any} builder */
```
Comment (Rich-Harris) [inline packages/adapter-cloudflare-workers/index.js:70]: this is a bit cheeky but i'm not sure if it's worth making `Builder` public

File: packages/adapter-netlify/index.js [hunk 2]
```diff
@@ -153,7 +153,7 @@ function get_netlify_config() {
 
 /**
  * @param {NetlifyConfig} netlify_config
- * @param {import('@sveltejs/kit').Builder} builder
+ * @param {any} builder
```
Comment (Rich-Harris) [inline packages/adapter-netlify/index.js:156]: see https://github.com/sveltejs/kit/pull/4220/files#r819806068

File: packages/adapter-cloudflare-workers/index.js (hunk 1 above)
Comment (benmccann) [reply packages/adapter-cloudflare-workers/index.js:70]: The only thing we gain by not making it public is that we can rename it from `Builder` to something else, but we can't change any of the methods on it in any manner, so the API is already 95% locked. Personally, I would be in favor of making it public

Comment (Rich-Harris) [reply packages/adapter-cloudflare-workers/index.js]: true. updated

**Review by benmccann — APPROVED** [review]

**Pull request merged**